src/identity/interaction/email-password/storage/BaseAccountStore.ts

```diff
@@ -0,0 +1,122 @@
+import assert from 'assert';
+import { hash, compare } from 'bcrypt';
+import { v4 } from 'uuid';
+import type { KeyValueStorage } from '../../../../storage/keyvalue/KeyValueStorage';
+import { trimTrailingSlashes } from '../../../../util/PathUtil';
+import type { AccountStore } from './AccountStore';
+
+/**
+ * A payload to persist a user account
+ */
+export interface AccountPayload {
+  webId: string;
+  email: string;
+  password: string;
+}
+
+/**
+ * A payload to persist the fact that a user
+ * has requested to reset their password
+ */
+export interface ForgotPasswordPayload {
+  email: string;
+  recordId: string;
+}
+
+export type EmailPasswordData = AccountPayload | ForgotPasswordPayload;
+
+export interface BaseAccountStoreArgs {
+  storageName: string;
+  storage: KeyValueStorage<string, EmailPasswordData>;
+  saltRounds: number;
+}
+
+/**
+ * A EmailPasswordStore that uses a KeyValueStorage
+ * to persist its information.
+ */
+export class BaseAccountStore implements AccountStore {
+  private readonly storageName: string;
+  private readonly storage: KeyValueStorage<string, EmailPasswordData>;
+  private readonly saltRounds: number;
+
+  public constructor(args: BaseAccountStoreArgs) {
+    this.storageName = trimTrailingSlashes(args.storageName);
```

Comment:
There's no real reason to have this trimTrailingSlashes anymore since the keys no longer correspond to the identifier where the resource is stored. I guess I just kept it because I still wanted the strings to look nice in my head. :D